src/Nethermind/Nethermind.Facade/BlockchainBridge.cs

```diff
@@ -257,19 +254,32 @@ public CallOutput CreateAccessList(BlockHeader header, Transaction tx, Cancellat
                 transaction.Nonce = GetNonce(stateRoot, transaction.SenderAddress);
             }
 
-            BlockHeader callHeader = new(
-                blockHeader.Hash!,
-                Keccak.OfAnEmptySequenceRlp,
-                Address.Zero,
-                0,
-                treatBlockHeaderAsParentBlock ? blockHeader.Number + 1 : blockHeader.Number,
-                blockHeader.GasLimit,
-                timestamp,
-                Array.Empty<byte>());
-
-            callHeader.BaseFeePerGas = treatBlockHeaderAsParentBlock
-                ? BaseFeeCalculator.Calculate(blockHeader, _specProvider.GetSpec(callHeader.Number))
-                : blockHeader.BaseFeePerGas;
+            BlockHeader callHeader = treatBlockHeaderAsParentBlock
+                ? new(
+                    blockHeader.Hash!,
+                    Keccak.OfAnEmptySequenceRlp,
+                    Address.Zero,
+                    UInt256.Zero,
+                    blockHeader.Number + 1,
+                    blockHeader.GasLimit,
+                    Math.Max(blockHeader.Timestamp + 1, _timestamper.UnixTime.Seconds),
+                    Array.Empty<byte>())
+                {
+                    BaseFeePerGas = BaseFeeCalculator.Calculate(blockHeader, _specProvider.GetSpec(blockHeader.Number + 1)),
```

Comment:
worth noting that we are not setting mix hash here

Reply:
Yes but should we? Can we predict random value for future block?

Reply:
I think 0 is potentially an edge case for something from which you expect a random number, maybe some smart contract edge case bug will happen during estimateGas or ethCall because of random returning zero. I think we could for example set it to the same as the previous block random, or just generate a random number ourselves.

Reply:
I would go with generating a random number ourselves